reconcile/test/utils/test_gitlab_api.py

```diff
@@ -468,3 +470,70 @@ def test_get_group_members(
     assert mocked_gitlab_api.get_group_members("group") == [
         {"user": "small", "access_level": None}
     ]
+
+
+def test_share_project_with_group_positive(
+    mocker: MockerFixture, mocked_gitlab_api: GitLabApi
+):
+    project = create_autospec(Project)
+    project.members = create_autospec(ProjectMemberManager)
+    project.members.all.return_value = [
+        create_autospec(ProjectMember, id=mocked_gitlab_api.user.id, access_level=40)
+    ]
+    mocker.patch(
+        "reconcile.utils.gitlab_api.GitLabApi.get_project"
+    ).return_value = project
+    mocker.patch(
+        "reconcile.utils.gitlab_api.GitLabApi.get_access_level"
+    ).return_value = 40
```

Comment:
is this patch needed? same for later test cases

Reply:
if this patch is not added it treats the value of `get_access_level` as a `MagicMock` and not as `int`. As a result our comaprison at https://github.com/app-sre/qontract-reconcile/blob/64ea350e9a104f41c32458db9a4f43fdab072342/reconcile/utils/gitlab_api.py#L265 and 291 Fails. So i mocked this one seperately to return int

Reply:
that's because `gitlab` is being patched, here we shouldn't patch another `GitLabApi` method, since all those values just const, we can update import in `gitlab_api.py` to

```python
from gitlab.const import MAINTAINER_ACCESS
```

so this logic doesn't require patch